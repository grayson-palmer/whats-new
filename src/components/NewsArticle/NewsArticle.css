.news-card {
  background-color: #FFF;
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  justify-content: space-between;
  height: 70vh;
  margin: 0em 1em 1em;
  width: 45%;
}

img {
  height: 300px;
  width: 100%;
  object-fit: cover;
}

.footer-info {
  background-color: #B46AF1;
  width: 100%;
  font-size: 1.5em;
}

.arrow-img {
  height: auto;
  width: 4em;
}

.url-link {
  justify-content: space-around;
  display: flex;
  align-items: center;
  text-decoration: none;
  color: white;
}