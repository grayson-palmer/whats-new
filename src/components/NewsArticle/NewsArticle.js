import React from 'react';
import './NewsArticle.css';
import arrow from '../../arrow.png';

const NewsArticle = (props) => {
  return (
    <section className='news-card'>
      <section className='img-container'>
        <img src={props.img} alt={props.headline}/>
      </section>
      <h1 className='news-headline'>{props.headline}</h1>
      <p className='news-description'>{props.body}</p>
      <footer className='footer-info'>
        <a className='url-link' href={props.link}>Link To Article<img className='arrow-img' src={arrow} alt='arrow icon'/></a>
      </footer>
    </section>
  )
}

export default NewsArticle;