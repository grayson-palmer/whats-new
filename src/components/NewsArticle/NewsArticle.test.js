import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  it('should be an instance of component', () => {
    const wrapper = shallow(<NewsArticle 
      props = {[{
        headline: 'Thanos Back Again', 
        body: 'Purp dude is back', 
        img: 'imgPath',
        link: 'urlPath',
        key: 1
      }]}
    />)
    expect(wrapper).toMatchSnapshot();
  })
})