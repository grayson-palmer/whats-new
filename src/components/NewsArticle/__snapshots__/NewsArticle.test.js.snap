// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should be an instance of component 1`] = `
<section
  className="news-card"
>
  <section
    className="img-container"
  >
    <img />
  </section>
  <h1
    className="news-headline"
  />
  <p
    className="news-description"
  />
  <footer
    className="footer-info"
  >
    <a
      className="url-link"
    >
      Link To Article
      <img
        alt="arrow icon"
        className="arrow-img"
        src="arrow.png"
      />
    </a>
  </footer>
</section>
`;
